roles/openshift_health_checker/openshift_checks/etcd_volume_check.py

```diff
@@ -0,0 +1,68 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for determining information about etcd volume size.
+"""
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EtcdImageData(OpenShiftCheck):
+    """Check that etcd volume capacity is not over 50%"""
+
+    name = "etcd_volume_check"
+    tags = ["etcd_volume_check"]
+
+    def run(self, _, task_vars):
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        etcd_mountpath = self.get_etcd_mountpath(ansible_mounts)
+        etcd_diskspace_avail = self.get_etcd_diskspace(etcd_mountpath, ansible_mounts)
+        etcd_totaldiskspace = self.get_etcd_totaldiskspace(etcd_mountpath, ansible_mounts)
+
+        if float(etcd_totaldiskspace - etcd_diskspace_avail) / float(etcd_totaldiskspace) > 0.5:
+            return {"failed": True, "msg": "Your etcd volume is more than 50% full."}
+
+        return {"changed": False}
+
+    @staticmethod
+    def get_etcd_mountpath(ansible_mounts):
+        # priority list - in descending priority order
+        valid_etcd_mount_paths = ["/var/lib/etcd", "/var/lib", "/var", "/"]
+
+        if not ansible_mounts:
+            return None
+
+        etcd_mount_path_idx = None
```

Comment:
@sosiouxme Thanks, I like the idea of using a hashmap instead. I believe that even in that case, though, we would still need to loop through all available mount paths. If we find `/` first, we would not want to return `valid_etcd_mount_paths["/"]` right away, if a more specific `valid_etcd_mount_paths["/var/lib/etcd"]` also exists